Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.kt

```diff
@@ -0,0 +1,207 @@
+            .setNegativeButton(R.string.cancel_button_text) { dialog, which -> dismiss() }
+            .create()
+    }
+
+    private fun initializeViews(customLayout: View) {
+        specificFormatLayout =
+            customLayout.findViewById(R.id.pocketpaint_save_format_specific_options)
+        jpgView = inflater.inflate(
+            R.layout.dialog_pocketpaint_save_jpg_sub_dialog,
+            specificFormatLayout,
+            false
+        )
+        mySeekbar = jpgView.findViewById(R.id.pocketpaint_jpg_seekbar_save_info)
+        percentage = jpgView.findViewById(R.id.pocketpaint_percentage_save_info)
+        mySpinner = customLayout.findViewById(R.id.pocketpaint_save_dialog_spinner)
+        informationButton = customLayout.findViewById(R.id.pocketpaint_btn_save_info)
+        imageName = customLayout.findViewById(R.id.pocketpaint_image_name_save_text)
+    }
+
+    private fun initializeFunctioning() {
+        val spinnerArray = arrayListOf("png", "jpg", "ora")
+        val adapter =
+            ArrayAdapter(mySpinner.context, android.R.layout.simple_spinner_item, spinnerArray)
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        mySpinner.adapter = adapter
+        val percentageString = Integer.toString(FileIO.compressQuality)
+        percentage.text = percentageString
+        imageName.text = setName
+        mySeekbar.progress = FileIO.compressQuality
+        mySeekbar.setOnSeekBarChangeListener(this)
+        mySpinner.onItemSelectedListener = this
+        informationButton.setOnClickListener {
+            when {
+                FileIO.isCatrobatImage -> {
+                    presenter.showOraInformationDialog()
+                }
+                FileIO.compressFormat == Bitmap.CompressFormat.JPEG -> {
+                    presenter.showJpgInformationDialog()
+                }
+                else -> {
+                    presenter.showPngInformationDialog()
+                }
+            }
+        }
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+        when (parent.getItemAtPosition(position).toString()) {
+            "jpg" -> {
+                specificFormatLayout.removeAllViews()
+                specificFormatLayout.addView(jpgView)
+                FileIO.compressFormat = Bitmap.CompressFormat.JPEG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".jpg"
+            }
+            "png" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".png"
+            }
+            "ora" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = true
+                FileIO.ending = ".ora"
+            }
+        }
```

Comment:
```suggestion
        when (parent.getItemAtPosition(position).toString()) {
            "jpg" -> {
                foo(compressFormat=JPG, isCatrobatImage=false, ending=it)
                specificFormatLayout.addView(jpgView)
                
            }
            ....
        }
```
Please remove the duplicated code :) (please use a propper name instead of foo ;))

Reply:
done : )